src/plots/cartesian/axes.js

```diff
@@ -704,8 +711,11 @@ function arrayTicks(ax) {
     // except with more precision to the numbers
     if(!Array.isArray(text)) text = [];
 
+    // make sure showing ticks doesn't accidentally add new categories
+    var tickVal2l = ax.type === 'category' ? ax.d2l_noadd : ax.d2l;
```

Comment:
right, like https://github.com/plotly/plotly.js/pull/1191#discussion_r89206759 - there are a bunch of uses outside `makeCalcdata`, particularly in gl code that we need to sort out first.